src/components/pagination/PaginationComponent.tsx

```diff
@@ -0,0 +1,45 @@
+export default function PaginationComponent({ postsPerPage, totalPosts, paginate }: { postsPerPage: any, totalPosts: any, paginate: any }) {
+    const pageNumbers = [];
+    for (let index = 1; index < Math.ceil(totalPosts / postsPerPage); index++) {
```

Comment:
define this as a function outside the react function and use like this
`const pageNumbers = nameOfTheFunction(...)`